data/events/scripts/player.lua

```diff
@@ -93,13 +93,43 @@ function Player:onMoveCreature(creature, fromPosition, toPosition)
 	return true
 end
 
-function Player:onReport(message, position, category)
+function Player:onReportRuleViolation(targetName, reportType, reportReason, comment, translation)
+	local name = self:getName()
+	local pendingReport = function () local f = io.open(string.format("data/reports/players/%s-%s-%d.txt", name, targetName, reportType), "r") ; if f then io.close(f) return true else return false end end
+	if pendingReport() then
+		self:sendTextMessage(MESSAGE_EVENT_ADVANCE, "Your report is being processed.")
+		return
+	end
+
+	local file = io.open(string.format("data/reports/players/%s-%s-%d.txt", name, targetName, reportType), "a")
+	if not file then
+		self:sendTextMessage(MESSAGE_EVENT_ADVANCE, "There was an error when processing your report, please contact a gamemaster.")
+		return
+	end
+
+	io.output(file)
+	io.write("------------------------------\n")
+	io.write("Complainter: " .. name .. "\n")
+	io.write("Reported: " .. targetName .. "\n")
```

Comment:
Mmmm I'm unsure about words. I'd use `Reported by` for the reported and `Target` for the reported player